projects/graph/src/graph.py

```diff
@@ -146,6 +146,39 @@ def dft_recursive(self, starting_vertex_id, visited=None):
             if neighbor not in visited:
                 self.dft_recursive(neighbor, visited)
 
+    def dfs(self, starting_vertex_id, query):
+        # Create an empty queue
+        s = Stack()
+        # Create an empty set of visited vertices
+        visited = set()
+
+        # create a list for path for each vertex
+        path = []
+
+        # Put the starting vertex in our Queue
+        s.push(starting_vertex_id)
+        # assign path for this vertex
+        path.append(starting_vertex_id)
+        s.push(path)
+
+        while s.size() > 0:
+
+            # get next item and path for that item in queue
+            path = s.pop()
```

Comment:
As we discussed before, while it is fine to have a mixed-object array/list in Python, try to understand an implementation that involves arrays with only a single type of object throughout.  This way, you can apply the same algorithm when you have to use a language such as C that restricts arrays to single-object-type only.

Reply:
got it.  in the lecture, only path is used and I changed it so only path (which is list) is stored in queue